biz.aQute.bnd.maven/src/aQute/bnd/maven/lib/resolve/BndrunContainer.java

```diff
@@ -263,8 +265,22 @@ public void setRunrequiresFromProjectArtifact(Run run) {
 
 			Artifact artifact = project.getArtifact();
 
+			String bsn = artifact.getArtifactId();
+
+			try {
+				Domain domain = Domain.domain(artifact.getFile());
+
+				bsn = Optional.of(domain.getBundleSymbolicName())
```

Comment:
getBundleSymbolicName can return null. Are you counting on the NPE from `of` ending up in the catch block?